cypress/e2e/registration.cy.js

```diff
@@ -1,11 +1,41 @@
 /// <reference types='cypress' />
 
+const generateUser = require('../support/generateUser');
+
 describe('Student Registration page', () => {
   before(() => {
-
+    cy.visit('/automation-practice-form');
   });
 
-  it('', () => {
+  it('should fill in the registration form and submits it', () => {
+    const user = generateUser();
+
+    cy.get('#firstName').type(user.firstName);
+    cy.get('#lastName').type(user.lastName);
+    cy.get('#userEmail').type(user.email);
+    cy.get('#genterWrapper').find('label').contains(user.sex).click();
+    cy.get('#userNumber').type(user.phone);
+    cy.selectBirthDate(user.birthDate);
+    user.subjects.forEach((el) => {
+      cy.get('#subjectsInput').type(el);
+      cy.get('.subjects-auto-complete__menu-list')
+        .find('.subjects-auto-complete__option')
+        .contains(el)
+        .click();
+    });
+    user.hobbies.forEach((el) => {
+      cy.get('#hobbiesWrapper').find('label').contains(el).click();
+    });
+    cy.get('#currentAddress').type(user.address);
+    cy.selectRandomChild('#state');
```

Comment:
The function `cy.selectRandomChild` is used here, but it is not a standard Cypress command. Ensure that this function is defined in your custom commands or elsewhere in your project.